modules/openapi-generator/src/main/resources/swift5/libraries/urlsession/URLSessionImplementations.mustache

```diff
@@ -170,11 +171,13 @@ private var credentialStore = SynchronizedDictionary<Int, URLCredential>()
             credentialStore[dataTask.taskIdentifier] = credential
 
             dataTask.resume()
-
+{{^useVapor}}{{^usePromiseKit}}{{^useResult}}{{^useAsyncAwait}}
+            return dataTask{{/useAsyncAwait}}{{/useResult}}{{/usePromiseKit}}{{/useVapor}}
         } catch {
             apiResponseQueue.async {
                 completion(.failure(ErrorResponse.error(415, nil, nil, error)))
-            }
+            }{{^useVapor}}{{^usePromiseKit}}{{^useResult}}{{^useAsyncAwait}}
```

Comment:
@DavidHorvath I think this logic should be removed from here, and be handled in the api.mustache, as already is

Reply:
Yeah we don't need that, sorry, I left that in... I'll update in a minute

Reply:
No problem 🙂 Thanks for your help

Reply:
Updated